libp2p/network/swarm.py

```diff
@@ -87,17 +88,53 @@ def set_stream_handler(self, stream_handler: StreamHandlerFn) -> None:
         try:
             # Get peer info from peer store
             addrs = self.peerstore.addrs(peer_id)
-        except PeerStoreError:
-            raise SwarmException(f"No known addresses to peer {peer_id}")
+        except PeerStoreError as error:
+            raise SwarmException(f"No known addresses to peer {peer_id}") from error
 
-        if not addrs:
-            raise SwarmException(f"No known addresses to peer {peer_id}")
+        exceptions: List[SwarmException] = []
+
+        # Try all known addresses
+        for multiaddr in addrs:
```

Comment:
Since you add the case we can try all addresses, would you mind adding a test for it?

Reply:
Yup, will do, I'll add test cases for:
1. No addresses
2. Multiple addresses
3. Explicitly one address
4. One address (that doesn't work)
5. Multiple addresses (that don't work)